src/status_im/ens/core.cljs

```diff
@@ -29,6 +31,26 @@
     username
     (stateofus/subdomain username)))
 
+(handlers/register-handler-fx
+ :update-ens-tx-state
+ (fn [{:keys [db]} [_ new-state username custom-domain? tx-hash]]
+   {:db (assoc-in db [:ens/registrations tx-hash] {:state new-state
+                                                   :username username
+                                                   :custom-domain? custom-domain?})}))
+
+(handlers/register-handler-fx
+ :update-ens-tx-state-and-redirect
+ (fn [cofx [_ new-state username custom-domain? tx-hash]]
+   (fx/merge cofx
+             (re-frame/dispatch [:update-ens-tx-state new-state username custom-domain? tx-hash])
```

Comment:
This is dispatching straight away, `re-frame/dispatch` would immediately dispatch as called (is a side-effect).
We use `fx` to run side effects, so in your case it would simply be:
```
{:dispatch-n  [[:update-ens-tx-state new-state username custom-domain? tx-hash]
                     [::redirect-to-ens-summary]]}
```